dblookupext/miniblocksHandler.go

```diff
@@ -366,3 +366,53 @@ func hasTxHashInMiniblockMetadataOnBlock(mbInfo *MiniblockMetadataOnBlock, txHas
 
 	return false
 }
+
+func (mh *miniblocksHandler) updateMiniblockMetadataOnBlock(
+	miniblockHash []byte,
+	headerHash []byte,
+	updateHandler func(mbMetadataOnBlock *MiniblockMetadataOnBlock),
+) error {
+	epoch, err := mh.epochIndex.getEpochByHash(miniblockHash)
+	if err != nil {
+		return err
+	}
+
+	shouldContinue, err := mh.updateMiniblockMetadataOnBlockInEpoch(epoch, miniblockHash, headerHash, updateHandler)
+	if err != nil {
+		return err
+	}
+	if !shouldContinue {
+		return nil
+	}
+
+	shouldContinue, err = mh.updateMiniblockMetadataOnBlockInEpoch(epoch+1, miniblockHash, headerHash, updateHandler)
+	if err != nil {
+		return err
+	}
+	if shouldContinue {
```

Comment:
`if !shouldContinue`? 
if not, there is a slight inconsistency here ("if should continue, then return err not found in storage")

Reply:
Actually, this is correct. 
Reason: the `updateMiniblockMetadataOnBlockInEpoch` returns (false, nil) if we found a miniblock metadata that contained the blockhash. (true, nil) signals that on this epoch we did not find the required block hash and also we did not encounter any error. So the algorithm will try to check the next epoch. On that epoch, if we still do not find the block hash (the function still returns (true, nil)) we need to force an error as that part of the miniblock wasn't notarized yet.
The inner test `partial miniblock hash with invalid header hash should not call update` test this exact scenario.